app/models/recommendation.rb

```diff
@@ -24,4 +24,12 @@ class Recommendation < VersionedRecord
 
   validates :title, presence: true
   validates :reference, presence: true, uniqueness: true
+
+  def is_current
+    reporting_cycle_taxonomy = Taxonomy.joins(:taxonomy).where(has_date: true).where(taxonomies_taxonomies: {priority: 1}).first
+    return true if reporting_cycle_taxonomy.nil? || categories.none? { _1.taxonomy_id == reporting_cycle_taxonomy.id }
+
+    current_category = reporting_cycle_taxonomy.categories.order(Arel.sql("COALESCE(date, created_at) DESC")).first
```

Comment:
the categories should be checked like this for every parent category (human rights body)

Reply:
not sure about `created_at` here though. it should only depend on the `date` value (could exclude those without a date)

Reply:
i think what's missing though is that the `reporting_cycle_taxonomy.categories` should be limited to the ones that are of the recommendation's human rights body (through its reporting cycle). otherwise we will be sorting (and mixing) all reporting cycles from all HR bodies